src/lexer/tokens.js

```diff
@@ -41,10 +42,16 @@ module.exports = {
       while (ch == " ") {
         ch = this.input();
       }
+      let isEnum = false;
+      if (this.is_LABEL_START()) {
+        while (this.is_LABEL()) {
+          ch += this.input();
+        }
+        isEnum = !ch.startsWith("extends") && !ch.startsWith("implements");
```

Comment:
Why `startsWith` rather than `!==`? I think you need to check for equality to cover e.g. `enum extendsType`. And come to think of it, I think you also need `ch.toLowerCase()`, since this is perfectly valid:

```php
class Enum exTEnds Foo {}
```

Reply:
I was using startswith because `ch` contains one character too much (the first non-label character is appended in the last iteration. But you're right of course, `startsWith` is not what we need. Fixed, and also added `toLowerCase`.

Reply:
Aah, gotcha. Yeah, that makes sense. LGTM now!

Reply:
🎉